packages/fuels-programs/src/lib.rs

```diff
@@ -21,3 +32,323 @@ impl Configurables {
+            receipts: relevant_receipts,
+        }
+    }
+
+    /// Based on receipts returned by a script transaction, the contract ID (in the case of a contract call),
+    /// and the output param, parse the values and return them as Token.
+    pub fn try_parse_output(
+        &mut self,
+        contract_id: Option<&Bech32ContractId>,
+        output_param: &ParamType,
+    ) -> Result<Token> {
+        let null_contract_id = ContractId::new([0u8; 32]);
+        let contract_id: ContractId = match contract_id {
+            Some(contract_id) => contract_id.into(),
+            // During a script execution, the script's contract id is the **null** contract id
+            None => null_contract_id,
+        };
+
+        let encoded_value = match output_param.get_return_location() {
+            ReturnLocation::ReturnData if output_param.is_vm_heap_type() => {
+                self.extract_return_data_heap(contract_id)
+            }
+            ReturnLocation::ReturnData => self.extract_return_data(contract_id),
+            ReturnLocation::Return => self.extract_return(contract_id),
+        };
+
+        if let Some(value) = encoded_value {
+            //self.receipts.remove(index);
+            let decoded_value = ABIDecoder::decode_single(output_param, &value)?;
+            Ok(decoded_value)
+        } else {
+            Err(Self::parsing_error(output_param))
+        }
+    }
+
+    fn parsing_error(output_param: &ParamType) -> Error {
+        error!(
+            InvalidData,
+            "ReceiptDecoder: failed to find matching receipts entry for {:?}", output_param
+        )
+    }
+
+    fn extract_return_data(&mut self, contract_id: ContractId) -> Option<Vec<u8>> {
+        let (index, data) = self.receipts.iter().enumerate().find_map(|(i, receipt)| {
+            if matches!(receipt,
+                    Receipt::ReturnData { id, .. } if *id == contract_id)
+            {
+                Some((
+                    i,
+                    receipt
+                        .data()
+                        .expect("ReturnData should have data")
+                        .to_vec(),
+                ))
+            } else {
+                None
+            }
+        })?;
+
+        self.receipts.remove(index);
+        Some(data)
+    }
+
+    fn extract_return(&mut self, contract_id: ContractId) -> Option<Vec<u8>> {
+        let (index, data) = self.receipts.iter().enumerate().find_map(|(i, receipt)| {
+            if matches!(receipt,
+                    Receipt::Return { id, ..} if *id == contract_id)
+            {
+                Some((
+                    i,
+                    receipt
+                        .val()
+                        .expect("Return should have val")
+                        .to_be_bytes()
+                        .to_vec(),
+                ))
+            } else {
+                None
+            }
+        })?;
+
+        self.receipts.remove(index);
+        Some(data)
+    }
```

Comment:
```suggestion
    fn extract_return(&mut self, contract_id: &ContractId) -> Option<Vec<u8>> {
        for (index, receipt) in self.receipts.iter().enumerate() {
            if let Receipt::Return { id, val, .. } = receipt {
                if id == contract_id {
                    let data = val.to_be_bytes().to_vec();
                    self.receipts.remove(index);
                    return Some(data);
                }
            }
        }
        None
    }
```
Some of the points of the previous suggestion apply here as well:
* no double match on `Return`
* no handling of the impossible case that a `Return` isn't a `Return`.